test/e2e/apps/statefulset.go

```diff
@@ -918,6 +918,110 @@ var _ = SIGDescribe("AppStatefulSetStorage", func() {
 			sst.WaitForStatusPVCReadyReplicas(ss, 1)
 		})
 	})
+
+	ginkgo.Describe("Recreate Pod when modifying volumeClaimTemplate with OnPodRollingUpdate", func() {
```

Comment:
Please add cases about sts with`InPlaceUpdateReady`.

Reply:
Please add a case about rename volume name in vcts and volumeMounts.

Reply:
sry. 
By default, the StatefulSets created by `NewStatefulSet` use the Recreate update strategy. 
We should add test cases to cover all scenarios (adding, removing, and renaming) for StatefulSets with `InPlaceIfPossiblePodUpdateStrategyType` update strategy.